### PR TITLE
feat(parser): support top-level await

### PR DESCRIPTION
**Pull request by 3cp**

### CODE REVIEW

**Comment by 3cp** [comment]
Running into an issue. The meriyah `Context.InGlobal` is not same as kataw's `Context.TopLevel`.
Notably the parseParenthesizedExpression clears Context.InGlobal on context.

So right now, it still rejects top-level await if it's in a parenthesized expression. `const foo = (await bar);`

**Comment by KFlash** [comment]
Why that bitwise mask got unset? Try set it back and see if any tests breaks.

**Comment by KFlash** [comment]
Why that bitwise mask got unset? Try set it back and see if any tests breaks.

**Comment by 3cp** [comment]
I thought the same.
After I changed this line in parseParenthesizedExpression
```js
  // context = (context | Context.DisallowIn | Context.InGlobal) ^ (Context.InGlobal | Context.DisallowIn);
  context = (context | Context.DisallowIn) ^ Context.DisallowIn;
```

Strangely it now failed to reject this code `['({ def\\u0061ult })', Context.None],`, it should trigger an invalid escaped keyword "default".

**Comment by KFlash** [comment]
I think that is caused by the fact you have used 31 "slots" for context masks and it turns into negative value with side effects

**Comment by 3cp** [comment]
I thought `OptionsUniqueKeyInPattern = 1 << 31,` needs to be explicitly turned on to have that issue?

**Comment by KFlash** [comment]
it depends on TS. Rare issues happens with TS all the time when it comes to enum values. Anyway. Non-failing escape keyword test is ok vs current issue

**Comment by 3cp** [comment]
I agree, I will just block that test.

File: src/errors.ts [hunk 1]
```diff
@@ -274,7 +274,6 @@ export const errorMessages: {
   [Errors.IllegalReturn]: 'Illegal return statement',
   [Errors.InvalidForLHSBinding]: 'The left hand side of the for-header binding declaration is not destructible',
   [Errors.InvalidNewTarget]: 'new.target only allowed within functions',
-  [Errors.InvalidEscapedKeyword]: "'Unexpected token: 'escaped keyword'",
```
Comment (3cp) [inline src/errors.ts:277]: This is a duplicated entry of line 207: `[Errors.InvalidEscapedKeyword]: "Unexpected token: 'escaped keyword'"`

**Comment by 3cp** [comment]
Somehow the circleci tests are not showing in this PR checks.

**Comment by KFlash** [comment]
isn't that a circleci issue?

**Comment by 3cp** [comment]
No idea. I don't use circleci personally.

**Comment by KFlash** [comment]
just merge this as is ?

**Comment by 3cp** [comment]
Sure, if you found the code change is reasonable.

**Review by KFlash — APPROVED** [review]

**Review by KFlash — APPROVED** [review]

**Pull request merged**